.github/workflows/client-compatibility-tests.yml

```
@@ -38,11 +38,38 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  changes:
+    name: Find changed files
+    runs-on: ubuntu-latest
+    outputs:
+      not_ignored_changes: ${{ steps.filter.outputs.every_changed == 'true' || steps.filter.outputs.any_changed == 'true' }}
```

The double negative `not ignored` seems confusing here. How about something simpler like `has_changes`?
```suggestion
      has_changes: ${{ steps.filter.outputs.every_changed == 'true' || steps.filter.outputs.any_changed == 'true' }}
```